src/test/java/com/connexta/store/StoreITests.java

```diff
@@ -528,4 +470,124 @@ protected boolean matchesSafely(URI uri) {
       }
     };
   }
+
+  @Test
+  public void testSuccessfulIngestRequest() throws Exception {
+    AtomicReference<String> metacardId = new AtomicReference<>();
+    transformMockRestServiceServer
+        .expect(requestTo(endpointUrlTransform))
+        .andExpect(method(HttpMethod.POST))
+        .andExpect(header("Accept-Version", endpointsTransformVersion))
+        .andExpect(jsonPath("$.mimeType").value(MediaType.TEXT_PLAIN_VALUE))
+        .andExpect(
+            request -> {
+              final String metacardLocation =
+                  (new JsonPathExpectationsHelper("$.metacardLocation"))
+                          .evaluateJsonPath(
+                              ((MockClientHttpRequest) request).getBodyAsString(), String.class)
+                      + "/metacard";
+              metacardId.set(metacardLocation);
+              webTestClient
+                  .get()
+                  .uri(metacardLocation)
+                  .exchange()
+                  .expectStatus()
+                  .isOk()
+                  .expectHeader()
+                  .contentType(METACARD_MEDIA_TYPE)
+                  .expectBody(Resource.class)
+                  .value(isReadable())
+                  .value(hasContents(METACARD.getInputStream()));
+            })
+        .andRespond(
+            withStatus(HttpStatus.ACCEPTED)
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(
+                    new JSONObject()
+                        .put("id", "asdf")
+                        .put("message", "The ID asdf has been accepted")
+                        .toString()));
+
+    // when
+    webTestClient
+        .post()
+        .uri("/ingest")
+        .contentType(MediaType.MULTIPART_FORM_DATA)
+        .syncBody(TEST_INGEST_REQUEST_BODY)
+        .header(
+            "Accept-Version",
+            "0.5.0") // TODO inject this like we do for the transformApiVersion in TransformClient
+        .header(LAST_MODIFIED, LAST_MODIFIED_DATE)
+        .exchange()
+        .expectStatus()
+        .isAccepted();
+
+    String id[] = StringUtils.split(metacardId.get(), '/');
+    assertNotNull(amazonS3.getObject(metacardBucket, id[3]));
```

Comment:
What does `id[3]` represent? Consider assigning `id[3]` to a local var whose name shows what it represents.